lib/lightstep/tracer/span.rb

```diff
@@ -0,0 +1,130 @@
+# TODO(ngauthier@gmail.com) Separate Span and SpanContext under a getter according
+# to the spec. Baggage moves to span context.
+
+module LightStep
+  class Span
+    # ----------------------------------------------------------------------------
+    #  OpenTracing API
+    # ----------------------------------------------------------------------------
+
+    attr_reader :tracer
+
+    # TODO(ngauthier@gmail.com) validate value is string, bool, or number and
+    # remove value.to_s from all calling code
+    def set_tag(key, value)
+      @tags[key] = value
+      self
+    end
+
+    # FIXME(ngauthier@gmail.com) accessor
+    def set_baggage_item(key, value)
+      @baggage[key] = value
+      self
+    end
+
+    # FIXME(ngauthier@gmail.com) accessor
+    # TODO(ngauthier@gmail.com) remove? Not in spec to get baggage.
+    def get_baggage_item(key)
+      @baggage[key]
+    end
+
+    # TODO(ngauthier@gmail.com) remove, since it's deprecated
+    def log_event(event, payload = nil)
+      log(event: event.to_s, payload: payload)
+    end
+
+    def log(fields)
+      record = { span_guid: @guid.to_s }
+
+      record[:stable_name] = fields[:event].to_s unless fields[:event].nil?
+      unless fields[:timestamp].nil?
+        record[:timestamp_micros] = (fields[:timestamp] * 1000).to_i
+      end
+      @tracer.raw_log_record(record, fields[:payload])
```

Comment:
fine for now, but note to self that we eventually need to support keys other than timestamp and payload (which will just map into the raw_log_record's payload somehow)... this is the compatibility bridge we've built in other languages.